src/borg/testsuite/archiver.py

```diff
@@ -1298,9 +1298,12 @@ def test_list_size(self):
         self.cmd('init', self.repository_location)
         test_archive = self.repository_location + '::test'
         self.cmd('create', '-C', 'lz4', test_archive, 'input')
-        output = self.cmd('list', '--format', '{size} {csize} {path}{NL}', test_archive)
-        size, csize, path = output.split("\n")[1].split(" ")
+        output = self.cmd('list', '--format', '{size} {csize} {dsize} {dcsize} {path}{NL}', test_archive)
+        size, csize, dsize, dcsize, path = output.split("\n")[1].split(" ")
         assert int(csize) < int(size)
+        assert int(dcsize) < int(dsize)
+        assert int(dsize) <= int(size)
+        assert int(dcsize) <= int(csize)
```

Comment:
hmm, while thinking about the tests, I've found a problem with the semantics of d(c)size:

this shall indicate a "deduplicated size" (or, in other words, the size that this file adds to the repo via unique chunks).
but there is a problem with just looking at refcount == 1 chunks. there could be also refcount > 1 chunks that are **only** used within **this** file, so if we want to be totally correct, we would also have to add them to dsize.